aws/resource_aws_msk_cluster.go

```diff
@@ -611,6 +631,16 @@ func resourceAwsMskClusterDeleteWaiter(conn *kafka.Kafka, arn string) error {
 
 		return resource.RetryableError(fmt.Errorf("timeout while waiting for the cluster %q to be deleted", arn))
 	})
+	if isResourceTimeoutError(err) {
+		_, err = conn.DescribeCluster(input)
+		if err != nil && isAWSErr(err, kafka.ErrCodeNotFoundException, "") {
```

Comment:
The `err != nil` check is extraneous here 👍 

```suggestion
		if isAWSErr(err, kafka.ErrCodeNotFoundException, "") {
```